addons/account/models/account_bank_statement.py

```diff
@@ -170,303 +45,191 @@ def _get_move_line_count(self):
-            next_statement = self.search([('previous_statement_id', '=', statement.id), ('journal_id', '=', statement.journal_id.id)])
-            if next_statement:
-                next_statement.previous_statement_id = statement.previous_statement_id
-        return super(AccountBankStatement, self).unlink()
-
-    # -------------------------------------------------------------------------
-    # CONSTRAINT METHODS
-    # -------------------------------------------------------------------------
-
-    @api.constrains('journal_id')
-    def _check_journal(self):
-        for statement in self:
-            if any(st_line.journal_id != statement.journal_id for st_line in statement.line_ids):
-                raise ValidationError(_('The journal of a bank statement line must always be the same as the bank statement one.'))
-
-    def _constrains_date_sequence(self):
-        # Multiple import methods set the name to things that are not sequences:
-        # i.e. Statement from {date1} to {date2}
-        # It makes this constraint not applicable, and it is less needed on bank statements as it
-        # is only an indication and not some thing legal.
-        return
-
-    # -------------------------------------------------------------------------
-    # BUSINESS METHODS
-    # -------------------------------------------------------------------------
-
-    def open_cashbox_id(self):
-        self.ensure_one()
-        context = dict(self.env.context or {})
-        if context.get('balance'):
-            context['statement_id'] = self.id
-            if context['balance'] == 'start':
-                cashbox_id = self.cashbox_start_id.id
-            elif context['balance'] == 'close':
-                cashbox_id = self.cashbox_end_id.id
-            else:
-                cashbox_id = False
-
-            action = {
-                'name': _('Cash Control'),
-                'view_mode': 'form',
-                'res_model': 'account.bank.statement.cashbox',
-                'view_id': self.env.ref('account.view_account_bnk_stmt_cashbox_footer').id,
-                'type': 'ir.actions.act_window',
-                'res_id': cashbox_id,
-                'context': context,
-                'target': 'new'
-            }
-
-            return action
-
-    def button_post(self):
-        ''' Move the bank statements from 'draft' to 'posted'. '''
-        if any(statement.state != 'open' for statement in self):
-            raise UserError(_("Only new statements can be posted."))
-
-        self._check_cash_balance_end_real_same_as_computed()
-
-        for statement in self:
-            if not statement.name:
-                statement._set_next_sequence()
-
-        self.write({'state': 'posted'})
+        Currently it removes the statements with no lines, can be extended if needed
+        """
+        self.search([
+            ('line_ids', '=', False),
+            ('write_date', '<', fields.Date.context_today(self))
+        ]).unlink()
```

Comment:
```suggestion
        self.search([('line_ids', '=', False), ('write_date', '<', fields.Date.context_today(self))]).unlink()
```
Also, you need to compare the o2m with a list
```
Traceback (most recent call last):
  File "/usr/lib/python3.8/code.py", line 90, in runcode
    exec(code, self.locals)
  File "<console>", line 1, in <module>
  File "/home/odoo/src/odoo/odoo/models.py", line 1902, in search
    res = self._search(domain, offset=offset, limit=limit, order=order, count=count)
  File "/home/odoo/src/odoo/odoo/models.py", line 4888, in _search
    query = self._where_calc(domain)
  File "/home/odoo/src/enterprise/account_reports_cash_basis/models/account_report.py", line 108, in _where_calc
    query = super()._where_calc(domain, active_test)
  File "/home/odoo/src/odoo/odoo/models.py", line 4636, in _where_calc
    return expression.expression(domain, self).query
  File "/home/odoo/src/odoo/odoo/osv/expression.py", line 446, in __init__
    self.parse()
  File "/home/odoo/src/odoo/odoo/osv/expression.py", line 671, in parse
    raise ValueError("Invalid field %s.%s in leaf %s" % (model._name, path[0], str(leaf)))
ValueError: Invalid field account.move.line.line_ids in leaf ('line_ids', '=', False)
>>>
```
(tested on account.move.line)

`('write_date', '<', fields.Date.context_today(self))` looks useless here

Reply:
we erase them from yesterday, because users may want to remove them from one line and put it in another line